test/thirdparty.test.ts

```diff
@@ -0,0 +1,227 @@
+
+          // Using orderBy is a special case, so there's a few possible syntaxes for it
+          orderBy: { column: "id", descending: true }, // ORDER BY id DESC NULLS LAST
+        }
+      );
+
+      // Using the database directly
+      const stmt = db.prepare(query).bind(bindings);
+      const { results } = await stmt.all();
+      deepStrictEqual(results, [
+        {
+          name: "Bobby Tables",
+          id: 1,
+          email: "bobby-tab@gmail.com",
+        },
+      ]);
+    });
+
+    test("where upserts are easier when using an orm", async function () {
+      const user: User = {
+        id: 1,
+        name: "John Doe",
+        email: "john-doe@gmail.com",
+      };
+      const { query, bindings } = GenerateQuery<User>(
+        QueryType.UPSERT,
+        users.tableName, // Could also come from the above meta.txn objects
+        {
+          data: user,
+          upsertOnlyUpdateData: {
+            name: user.name,
+            email: user.email,
+          },
+          where: {
+            id: user.id,
+          },
+        },
+        "id"
+      );
+      // {
+      //    query: "INSERT INTO users (id, name, email) VALUES (?, ?, ?) ON CONFLICT(id) DO UPDATE SET name = ?, email = ? WHERE id = ?",
+      //    bindings: [1, "John Doe", "john-doe@gmail.com", "John Doe", "john-doe@gmail.com", 1]
+      // }
+
+      // Using the database directly
+      const stmt = db.prepare(query).bind(bindings);
+      const { meta } = await stmt.run();
+      const receipt = await (meta.txn as any).wait();
+      strictEqual(receipt?.error, undefined);
+
+      const results = await db
+        .prepare(`SELECT * FROM ${users.tableName} WHERE id=?`)
+        .bind(user.id)
+        .first<User>();
+      deepStrictEqual(results, user);
+    });
+  });
+
+  describe("@databases/sql", function () {
+    // See https://www.atdatabases.org/docs/sqlite
+    const sqliteFormat: FormatConfig = {
+      escapeIdentifier: (str) => escapeSQLiteIdentifier(str),
+      formatValue: (value) => ({ placeholder: "?", value }),
+    };
+    let tableName: string;
+
+    this.beforeAll(async function () {
+      this.timeout("10s");
+
+      // First, we'll test out using sql identifiers
+      const primaryKey = sql.ident("id");
+      const query = sql`CREATE TABLE test_sql (${primaryKey} integer primary key, counter integer, info text);`;
+      const { text, values } = query.format(sqliteFormat);
+      const { meta } = await db.prepare(text).bind(values).run();
+      const { name } = await meta.txn!.wait();
+      tableName = name!;
+    });
+
+    test("inserting rows with interpolated values is possible", async function () {
+      const one = 1;
+      const four = 4;
+      const three = sql.value("three");
+      // Here's a safer way to inject table names
+      const query = sql`INSERT INTO ${sql.ident(
+        tableName
+      )} (counter, ${sql.ident("info")})
```

Comment:
Here's where this pattern shines. We can _safely_ interpolate identifiers into a query string using this library. I'm actually half tempted to do this in our own source using this lib. This is much nicer than just string interpolation, and safer because it'll escape properly. The same goes for values, and just about anything else you interpolate into the tagged template literal. It remain to be seen if we can auto-encode things like dates and blobs... I think it'll work.